app/models/runtime/deployment_model.rb

```diff
@@ -99,6 +114,27 @@ def continuable?
       state == DeploymentModel::PAUSED_STATE
     end
 
+    def canary_step
+      raise 'canary_step is only valid for canary deloyments' unless strategy == CANARY_STRATEGY
```

Comment:
*note* Calling this is a programming error so seemed best to raise an exception.